internal/oci/container.go

```diff
@@ -163,7 +175,36 @@ func (c *Container) FromDisk() error {
 	defer jsonSource.Close()
 
 	dec := json.NewDecoder(jsonSource)
-	return dec.Decode(c.state)
+	tmpState := &ContainerState{}
+	if err := dec.Decode(tmpState); err != nil {
+		return err
+	}
+
+	// this is to handle the situation in which we're upgrading
+	// versions of cri-o, and we didn't used to have this information in the state
+	if tmpState.InitPid == 0 && tmpState.InitStartTime == "" && tmpState.Pid != 0 {
+		if err := tmpState.SetInitPid(tmpState.Pid); err != nil {
+			return err
+		}
+		logrus.Infof("pid information for container %s updated to %d %s", c.id, tmpState.InitPid, tmpState.InitStartTime)
+	}
+	c.state = tmpState
+	return nil
+}
+
+// SetInitPid initializes the InitPid and InitStartTime for the container state
+// given a pid.
+// These values should be set once, and not changed again.
+func (cstate *ContainerState) SetInitPid(pid int) (err error) {
+	if cstate.InitPid != 0 || cstate.InitStartTime != "" {
+		return errors.Errorf("pid and start time already initialized: %d %s", cstate.InitPid, cstate.InitStartTime)
+	}
+	cstate.InitPid = pid
+	cstate.InitStartTime, err = getInitStartTime(pid)
+	if err != nil {
```

Comment:
This is useless, since we use a named return value (`err`) and it is already initialized.

Note: in the spirit of that recent PR that renames `err` to `retErr` I would use `retErr` here as well (or not use a named return value at all -- up to you).